### DISCUSSION TITLE
Adding an email button.

### DISCUSSION THREAD

**Discussion by joelvargheseg** [stephanrauh/ngx-extended-pdf-viewer#1200] [Ideas]
Hello developers,
I was just wondering is there anyway you can integrate an email functionality in this viewer. so previously I was using Adobe plugin for my project which had an email functionality which basically opens the client side email tool and will pass the PDF as an attachment. but it only worked in internet explorer and my solution was to use ngx-extended-pdf-viewer which can be used across any platform.
Right now I was trying to imitate that email functionality using my angular-java web project and its kinda impossible to open local client side default mail app (specifically outlook) from my Angular UI. 
or if you found a way to implement this functionality from your UI please can you share the resources which I can also follow?
thank you.

**Comment by stephanrauh** [comment]
How do you want to trigger the email functionality? Does it suffice to add a new button or menu item that opens the email application?

If so, you probably can add the email functionality yourself. You can customize the toolbars.  In particular, you can add your own menu items or buttons. Remains the question how to open the email client from Angular and how to pass an email content to it. I haven't researched the latter topic yet, but StackOverflow says it's possible: https://stackoverflow.com/questions/13231125/automatically-open-default-email-client-and-pre-populate-content

**Comment by joelvargheseg** [comment]
I just wanted to have a button with an email icon so when I press it, it automatically opens outlook create email and attach the document with it.

**Comment by joelvargheseg** [comment]
currently mailTo as described in the stackoverflow won't allow to send an attachment through UI. I just wanted that functionality.
I tried many packages everything directly sends the email. I have no control over it.

**Reply by stephanrauh** [reply]
It seems that's not possible from JavaScript, and for a good reason. If the browser was allowed to send files from your hard disk without asking you for permission, it'd be hacker's paradise. So I guess the best alternative is to send a link to the PDF file:

https://mailtrap.io/blog/mailto-links-explained/

https://stackoverflow.com/questions/8657888/press-button-start-native-email-program-with-attachment-located-on-webserver

**Reply by joelvargheseg** [reply]
that's why I was asking about any way the pdf-viewer can send the data to the default mail app. no need for downloading the pdf.
